Sources/web3swift/EthereumABI/ABIDecoding.swift

```diff
@@ -173,7 +173,25 @@ extension ABIDecoder {
                 let (v, c) = decodeSignleType(type: subTypes[i], data: elementItself, pointer: consumed)
                 guard let valueUnwrapped = v, let consumedUnwrapped = c else {return (nil, nil)}
                 toReturn.append(valueUnwrapped)
-                consumed = consumed + consumedUnwrapped
+                /*
+                 When decoding a tuple that is not static or an array with a subtype that is not static, the second value in the tuple returned by decodeSignleType is a pointer to the next element, NOT the length of the consumed element. So when decoding such an element, consumed should be set to consumedUnwrapped, NOT incremented by consumedUnwrapped.
+                 */
+                switch subTypes[i] {
+                case .array(type: let subType, length: _):
+                    if !subType.isStatic {
+                        consumed = consumedUnwrapped
+                    } else {
+                        fallthrough
```

Comment:
do not use fallthrough. in this case it will drop down to the next case, and it looks for `!subTypes[i].isStatic` which as it was an array will process as true, and do `consumed = consumedUnwrapped`  instead of falling down to the default and doing `consumed = consumed + consumedUnwrapped` as intended.

As a general rule, don't ever use `fallthrough` unless absolutely necessary, and you fully understand all the potential side-effects